src/commcare_cloud/fab/operations/staticfiles.py

```diff
@@ -97,18 +97,57 @@ def collectstatic(use_current_release=False):
 
 
 @parallel
-@roles(ROLES_STATIC)
+@roles(ROLES_STATIC_PRIMARY)
 def compress(use_current_release=False):
     """Run Django Compressor after a code update"""
     venv = env.py3_virtualenv_root if not use_current_release else env.py3_virtualenv_current
     with cd(env.code_root if not use_current_release else env.code_current):
         sudo('{}/bin/python manage.py compress --force -v 0'.format(venv))
         sudo('{}/bin/python manage.py purge_compressed_files'.format(venv))
-    update_manifest(save=True, use_current_release=use_current_release)
+
+    push_manifest(use_current_release=use_current_release)
+
+
+def push_manifest(use_current_release=False):
+    if env.use_shared_dir_for_staticfiles:
+        with cd(env.code_root if not use_current_release else env.code_current):
+            git_hash = sudo('git rev-parse HEAD').strip()
```

Comment:
could this (and the stuff below), be pulled into a function, or even work by passing an argument to the other function. it feels weird to mix fab operations and raw shell commands like this